rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -183,66 +170,155 @@ def _query_blockchain_info_has_transactions(
-                self.database.set_dynamic_cache(
-                    write_cursor=write_cursor,
-                    name=DBCacheDynamic.LAST_BITCOIN_TX_TS,
-                    value=tx_list[0].timestamp,  # first in the list is the latest tx
-                    address=account,
-                )
+            block_height = tx_list[0].block_height if len(tx_list) > 0 else last_queried_block
+            new_block_height = max(new_block_height, block_height)
+
+        return new_block_height, tx_list
+
+    def _query_blockchain_info_transactions(
+            self,
+            accounts: Sequence[BTCAddress],
+            options: dict[str, Any],
+    ) -> tuple[int, list[BitcoinTx]]:
+        """Query blockchain.info for transactions.
+
+        We use the single address /rawaddr endpoint rather than the multiaddr endpoint since
+        some types of TxIOs are omitted from the multiaddr endpoint.
+
+        Returns a tuple containing the latest queried block height and the list of txs.
+        """
+        results: dict[BTCAddress, list[dict[str, Any]]] = defaultdict(list)
+        for account in accounts:
+            base_url = f'{BLOCKCHAIN_INFO_BASE_URL}/rawaddr/{account}'
+            offset, limit = 0, 50
+            while True:
+                results[account].extend(chunk := request_get_dict(
+                    url=f'{base_url}?n={limit}&offset={offset}',
+                    handle_429=True,
+                    backoff_in_seconds=10,
+                )['txs'])
+                if len(chunk) < limit:
+                    break  # all txs have been queried
+
+                offset += limit
+
+        return self._process_accounts_raw_tx_lists(
+            accounts_raw_tx_lists=results,
+            options=options,
+            processing_fn=BitcoinTx.deserialize_from_blockchain_info,
+        )
 
-        return tx_list
+    def _process_raw_tx_from_blockcypher(
+            self,
+            data: dict[str, Any],
+    ) -> BitcoinTx:
+        """Convert a raw tx dict from blockcypher into a BitcoinTx.
+        If the tx has a large number of TxIOs, the remaining TxIOs will be queried using the urls
+        provided in the API response.
+        May raise DeserializationError, KeyError, RemoteError, UnableToDecryptRemoteData.
+        """
+        inputs: list[dict[str, Any]] = []
+        outputs: list[dict[str, Any]] = []
+        for side, tx_io_list in (('inputs', inputs), ('outputs', outputs)):
+            next_data = data.copy()
+            while True:
+                tx_io_list.extend(list_chunk := next_data[side])
+                if (
+                    (next_url := next_data.get(f'next_{side}')) is None or
+                    len(list_chunk) < BLOCKCYPHER_TX_IO_LIMIT
+                ):
+                    break  # all TxIOs for this side have been queried
+
+                next_data = request_get_dict(url=next_url, handle_429=True, backoff_in_seconds=1)
+
+        data['inputs'] = inputs
```

Comment:
you are modifying the passed data. This is a very bad thing to do, especially without either the docstring or the name of the function specifying this is being done.

Why do you need to modify the data? If you do then return the modified data or generally change how this function works to avoid antipatterns like this